src/run_throttlebot.py

```diff
@@ -361,61 +374,92 @@ def run(system_config, workload_config, filter_config, default_mr_config):
             increment_to_performance = {}
             current_mr_allocation = resource_datastore.read_mr_alloc(redis_db, mr)
             print 'Current MR allocation is {}'.format(current_mr_allocation)
+            
             for stress_weight in stress_weights:
-                new_alloc = convert_percent_to_raw(mr, current_mr_allocation, stress_weight)
-                resource_modifier.set_mr_provision(mr, new_alloc)
+                # Calculate Gradient Schedule and provision resources accordingly
+                mr_gradient_schedule = calculate_mr_gradient_schedule(redis_db, [mr],
+                                                                      sys_config,
+                                                                      stress_weight)
+                for change_mr in mr_gradient_schedule:
+                    resource_modifier.set_mr_provision(change_mr, mr_gradient_schedule[change_mr])
+                    
                 experiment_results = measure_runtime(workload_config, experiment_trials)
                 
                 # Write results of experiment to Redis
                 # preferred_results = remove_outlier(experiment_results[preferred_performance_metric])
                 preferred_results = experiment_results[preferred_performance_metric]
                 mean_result = mean_list(preferred_results)
-                tbot_datastore.write_redis_ranking(redis_db, experiment_count, preferred_performance_metric, mean_result, mr, stress_weight)
-
-                # Remove the effect of the resource stressing
-                new_alloc = convert_percent_to_raw(mr, current_mr_allocation, 0)
-                resource_modifier.set_mr_provision(mr, new_alloc)
+                tbot_datastore.write_redis_ranking(redis_db, experiment_count,
+                                                   preferred_performance_metric,
+                                                   mean_result, mr, stress_weight)
+
+                # Revert the Gradient schedule and provision resources accordingly
+                mr_revert_gradient_schedule = revert_mr_gradient_schedule(redis_db, [mr], sys_config, stress_weight)
+                for change_mr in mr_revert_gradient_schedule:
+                    resource_modifier.set_mr_provision(change_mr, mr_revert_gradient_schedule[change_mr])
+                    
                 increment_to_performance[stress_weight] = experiment_results
 
             # Write the results of the iteration to Redis
             tbot_datastore.write_redis_results(redis_db, mr, increment_to_performance, experiment_count, preferred_performance_metric)
             print '*' * 20
             print '\n' * 2
 
-        # Getting experiment time
+        # Timing Information for the purpose of experiments
         current_time_stop = datetime.datetime.now()
         time_delta = current_time_stop - time_start
-
-        # Incrementing cumulative MR count
         cumulative_mr_count += len(mr_to_stress)
```

Comment:
Should be += len(mr_to_consider) since the list was renamed.

Reply:
Another line that I can't comment is L517. The variable new_alloc no longer exists so it should be updated.